boa_engine/src/object/internal_methods/mod.rs

```diff
@@ -25,6 +30,40 @@ pub(super) mod string;
 
 pub(crate) use array::ARRAY_EXOTIC_INTERNAL_METHODS;
 
+#[derive(Debug)]
+pub(crate) struct InternalMethodContext<'ctx, 'host> {
+    context: &'ctx mut Context<'host>,
+    slot: Slot,
+}
```

Comment:
Nice! By the way, this could implement `From<Context>` to simplify calls further e.g. `obj.__get__(key, target, &mut context.into())?`